kubernetes/schema_pod_spec.go

```diff
@@ -344,6 +344,30 @@ func podSpecFields(isUpdatable, isComputed bool) map[string]*schema.Schema {
 							Type: schema.TypeInt,
 						},
 					},
+					"windows_options": {
+						Type:        schema.TypeSet,
+						Description: "The Windows specific settings applied to all containers. If unspecified, the options within a container's SecurityContext will be used. If set in both SecurityContext and PodSecurityContext, the value specified in SecurityContext takes precedence. Note that this field cannot be set when spec.os.name is linux.",
+						Optional:    true,
+						Elem: &schema.Resource{
+							Schema: map[string]*schema.Schema{
+								"gmsa_credential_spec": {
+									Type:        schema.TypeString,
+									Description: "GMSACredentialSpec is where the GMSA admission webhook inlines the contents of the GMSA credential spec named by the GMSACredentialSpecName field",
+									Required:    true,
```

Comment:
How do you know this field and the one below are required? I don't see them marked as required in the K8s API docs.


edit: Looking at the type for that field, all the attributes seem to be optional: 

https://github.com/kubernetes/kubernetes/blob/master/pkg/apis/core/types.go#L5941-L5965

Reply:
@jrhouston this was overlooked on my part but I found the reason why I had marked it as `Required` in the first place. When using the `windows_options` block as optional and using it, we are given this error when it isn't used:

```
│ Error: Pod "terraform-example" is invalid: [spec.securityContext.windowsOptions.gmsaCredentialSpecName: Invalid value: "": a lowercase RFC 1123 subdomain must consist of lower case alphanumeric characters, '-' or '.', and must start and end with an alphanumeric character (e.g. 'example.com', regex used for validation is '[a-z0-9]([-a-z0-9]*[a-z0-9])?(\.[a-z0-9]([-a-z0-9]*[a-z0-9])?)*'), spec.securityContext.windowsOptions.gmsaCredentialSpec: Invalid value: "": gmsaCredentialSpec cannot be an empty string]
```

leading to the need for it to be `Required` instead of `Optional` since we can't just leave it as empty for both `gmsaCredentialSpec` and `gmsaCredentialSpecName`

I'll make this update in an upcoming PR that adds `windows_options` to the container block also.